source/exe/main_common.cc

```diff
@@ -81,6 +84,28 @@ MainCommonBase::MainCommonBase(OptionsImpl& options) : options_(options) {
 
 MainCommonBase::~MainCommonBase() { ares_library_cleanup(); }
 
+void MainCommonBase::configureComponentLogLevels() {
+  const std::vector<std::pair<std::string, std::string>>& component_log_levels =
+      options_.componentLogLevels();
+  for (auto& component_log_level : component_log_levels) {
+    const std::string& component = component_log_level.first;
+    const std::string& log_level = component_log_level.second;
+    size_t level_to_use = std::numeric_limits<size_t>::max();
+    for (size_t i = 0; i < ARRAY_SIZE(spdlog::level::level_names); i++) {
+      if (log_level == spdlog::level::level_names[i]) {
+        level_to_use = i;
+        break;
+      }
+    }
+    for (Logger::Logger& logger : Logger::Registry::loggers()) {
```

Comment:
What if the logger is not found? Can we fail on that with an error message? Same comment about factoring out into a utility function.